src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -496,6 +496,8 @@ private static ScriptType ParseScriptType(string scriptFilePath)
 
         internal Collection<FunctionDescriptor> ReadFunctions(List<FunctionMetadata> metadatas, IEnumerable<FunctionDescriptorProvider> descriptorProviders)
         {
+            metadatas = ApplyMaximumFunctionCount(metadatas);
```

Comment:
Maybe a bit simpler to ignore at the source in the `ScriptHost.ReadFunctions` loop?